jib-core/src/main/java/com/google/cloud/tools/jib/configuration/PortsWithProtocol.java

```diff
@@ -79,6 +79,30 @@ public Protocol getProtocol() {
     return protocol;
   }
 
+  @Override
+  public boolean equals(Object other) {
```

Comment:
btw where is this used (and hashCode)?

Reply:
It's used in tests for assertions, and hashCode was only added to avoid gradle warnings.

Reply:
Ah okay, if it's only for tests, it should just be a check in test files. having equals on a class is adding a lot of complexity and possibilities for error, so we should only use it if absolutely necessary.

Reply:
Is this going to be a user-facing class?

Reply:
If it's a user-facing class would it be nice to have?

Reply:
Hmm, okay, I guess that may be useful for users.